block/internal/syncing/syncer.go

```diff
@@ -262,107 +251,118 @@ func (s *Syncer) processLoop() {
-	logger.Info().Msg("starting P2P notifier worker")
-	defer logger.Info().Msg("P2P notifier worker stopped")
+	s.logger.Info().Msg("starting DA worker")
+	defer s.logger.Info().Msg("DA worker stopped")
+
+	nextDARequestAt := &time.Time{}
+	pollInterval := fastDAPollInterval
 
 	for {
+		s.tryFetchFromDA(nextDARequestAt)
 		select {
 		case <-s.ctx.Done():
 			return
-		case evt, ok := <-sub.C:
-			if !ok {
-				return
-			}
-			if evt.Type != expected {
-				logger.Debug().Str("received_type", string(evt.Type)).Msg("ignoring notifier event with unexpected type")
-				continue
-			}
-			s.logger.Debug().Str("event_type", string(evt.Type)).Str("source", string(evt.Source)).Uint64("height", evt.Height).Msg("received store event")
-			s.tryFetchFromP2P()
+		case <-time.After(pollInterval):
 		}
 	}
 }
 
-func (s *Syncer) daWorkerLoop() {
+func (s *Syncer) pendingWorkerLoop() {
 	defer s.wg.Done()
 
 	if !s.waitForGenesis() {
 		return
 	}
 
-	s.logger.Info().Msg("starting DA worker")
-	defer s.logger.Info().Msg("DA worker stopped")
+	s.logger.Info().Msg("starting pending worker")
+	defer s.logger.Info().Msg("pending worker stopped")
 
-	nextDARequestAt := &time.Time{}
-	pollInterval := min(10*time.Millisecond, s.config.Node.BlockTime.Duration)
-	if pollInterval <= 0 {
-		pollInterval = 10 * time.Millisecond
-	}
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
 
 	for {
-		s.tryFetchFromDA(nextDARequestAt)
 		select {
 		case <-s.ctx.Done():
 			return
-		case <-time.After(pollInterval):
+		case <-ticker.C:
+			s.processPendingEvents()
 		}
 	}
 }
 
-func (s *Syncer) pendingWorkerLoop() {
+func (s *Syncer) p2pWorkerLoop() {
 	defer s.wg.Done()
 
 	if !s.waitForGenesis() {
 		return
 	}
 
-	s.logger.Info().Msg("starting pending worker")
-	defer s.logger.Info().Msg("pending worker stopped")
-
-	ticker := time.NewTicker(10 * time.Millisecond)
-	defer ticker.Stop()
+	logger := s.logger.With().Str("worker", "p2p").Logger()
+	logger.Info().Msg("starting P2P worker")
+	defer logger.Info().Msg("P2P worker stopped")
 
 	for {
 		select {
 		case <-s.ctx.Done():
 			return
-		case <-ticker.C:
-			s.processPendingEvents()
+		default:
+		}
+
+		currentHeight, err := s.store.Height(s.ctx)
+		if err != nil {
+			logger.Error().Err(err).Msg("failed to get current height for P2P worker")
+			if !s.sleepOrDone(50 * time.Millisecond) {
+				return
+			}
+			continue
+		}
+
+		targetHeight := currentHeight + 1
+		waitCtx, cancel := context.WithCancel(s.ctx)
+		s.setP2PWaitState(targetHeight, cancel)
+
+		err = s.p2pHandler.ProcessHeight(waitCtx, targetHeight, s.heightInCh)
+		s.clearP2PWaitState(targetHeight)
+		cancel()
+
+		if err != nil {
+			if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+				continue
+			}
+
+			if waitCtx.Err() == nil {
+				logger.Warn().Err(err).Uint64("height", targetHeight).Msg("P2P handler failed to process height")
```

Comment:
I would assume that we run into this quite often when the height is not available, yet. IMHO it is worth to ignore the not found error

Reply:
if a height it not available then we block on waiting for it. should not get hit unless we have a different issue.